Update documentation for creating a controluser

ibennetch opened the pull request:
This now provides information for newer MySQL (newer than 8.0) and MariaDB servers.

Fixes: #15606

Signed-off-by: Isaac Bennetch <bennetch@gmail.com>

ibennetch commented:
I think I've gotten all of the scenarios correct here, but let me know if I missed some variation.

williamdes commented on doc/setup.rst:
```diff
@@ -788,9 +788,26 @@ You will also need to have a controluser
 
+For MySQL 8.0 and newer:
+
+.. code-block:: mysql
```
Can you mention the php version restriction?

ibennetch replied:
I'm not sure that PHP versions are relevant to creating the controluser. Besides, we already mention the PHP versions in the Requirements section, don't we?

williamdes replied:
I would add a line so users can understand why it does not work when they use sha2 method

sha2 method can be used from a non sha2 user, the user does not understand that the control user will not be able to connect because the method is not supported

Maybe my sentence is not clear enough, I do not know how to explain well the issue

mauriciofauth replied:
Isn't it better to have that information on a FAQ? And a link to the FAQ instead of repeating that info in various places.

ibennetch replied:
Proposed solution for the version problem is #15750.

ibennetch commented:
Oh yes, now it is quite clear.

ibennetch committed:
Update documentation for creating a controluser.

This now provides information for newer MySQL (newer than 8.0) and MariaDB servers.

Signed-off-by: Isaac Bennetch <bennetch@gmail.com>

williamdes reviewed (approved):
Thank you

ibennetch commented:
Thanks for the suggestions!
